innovation_sweet_spots/pipeline/pilot/investment_predictions/create_dataset.py

```diff
@@ -0,0 +1,167 @@
+            evaluation period for assessing companies. Defaults to "01/01/2014".
+        window_end_date: End date for window that simulates the evaluation period
+            for assessing companies. Defaults to "01/01/2018".
+    """
+    # Date information
+    window_start_date = pd.to_datetime(window_start_date)
+    window_end_date = pd.to_datetime(window_end_date)
+    success_start_date = window_end_date
+    pilot_outputs = PROJECT_DIR / "outputs/finals/pilot_outputs/"
+    # Load datasets
+    cb_orgs = pd.read_csv(pilot_outputs / "ISS_pilot_Crunchbase_companies.csv")
+    cb_acquisitions = get_crunchbase_acquisitions()
+    cb_ipos = get_crunchbase_ipos()
+    cb_funding_rounds = get_crunchbase_funding_rounds()
+
+    # Binarise columns
+    cols_to_binarise = [
+        "twitter_url",
+        "email",
+        "phone",
+        "facebook_url",
+        "homepage_url",
+        "linkedin_url",
+    ]
+    for col in cols_to_binarise:
+        cb_orgs = utils.convert_col_to_has_col(df=cb_orgs, col=col, drop=True)
+
+    # Select relevant cols
+    (
+        cb_orgs[
+            [
+                "id",
+                "name",
+                "legal_name",
+                "description",
+                "location_id",
+                "tech_category",
+                "has_email",
+                "has_phone",
+                "has_facebook_url",
+                "has_twitter_url",
+                "has_homepage_url",
+                "has_linkedin_url",
+                "founded_on",
+                "closed_on",
+            ]
+        ]
+        # Add additional variables
+        .pipe(utils.tech_cats_to_dummies)
+        .pipe(utils.dedupe_descriptions)
+        .pipe(utils.add_acquired_on, cb_acquisitions)
+        .pipe(utils.add_went_public_on, cb_ipos)
+        .pipe(utils.add_funding_round_ids, cb_funding_rounds)
+        .pipe(utils.add_funding_round_dates, cb_funding_rounds)
+        # Add flags for company founded/acquired/went public in time window
+        .pipe(
+            utils.window_flag,
+            start_date=window_start_date,
+            end_date=window_end_date,
+            variable="founded_on",
+        )
+        .pipe(
+            utils.window_flag,
+            start_date=window_start_date,
+            end_date=window_end_date,
+            variable="acquired_on",
+        )
+        .pipe(
+            utils.window_flag,
+            start_date=window_start_date,
+            end_date=window_end_date,
+            variable="went_public_on",
+        )
+        # Filter companies founded in the time window but not acquired or went public
+        .query(
+            "founded_on_in_window == 1 & acquired_on_in_window == 0 & went_public_on_in_window == 0"
+        )
+        .drop(
+            columns=[
+                "founded_on_in_window",
+                "acquired_on_in_window",
+                "went_public_on_in_window",
```

Comment:
Again, perhaps worth considering to define these at the top of of the script (as global "success" 'parameters') to make them more visible?

Reply:
I have moved the other column lists to the top as constants except these as I have slightly changed how this part of the code runs in the full UK dataset.